net/src/packet/mod.rs

```diff
@@ -50,7 +66,12 @@ impl<Buf: PacketBufferMut> Packet<Buf> {
         self.mbuf.take()
     }
 
-    pub(crate) fn reserialize(mut self) -> Buf {
+    /// Reserialize the packet into a buffer (consuming self).
+    ///
+    /// # Panics
+    ///
+    /// This method should never panic baring programmer error.
```

Comment:
> baring programmer error

This reads like the programmer using this function could make a mistake and trigger a `panic`, but I think you need it shouldn't panic unless we did a mistake in the function/class? So maybe “baring implementation error”? [Not blocking]